python/dgl/view.py

```diff
@@ -282,33 +282,75 @@ def __call__(self, ntype=None):
 
 class HeteroNodeDataView(MutableMapping):
     """The data view class when G.ndata[ntype] is called."""
-    __slots__ = ['_graph', '_ntype', '_ntid', '_nodes']
+    __slots__ = ['_graph', '_ntype', '_ntid', '_nodes', '_multi_ntype']
 
     def __init__(self, graph, ntype, ntid, nodes):
         self._graph = graph
         self._ntype = ntype
         self._ntid = ntid
         self._nodes = nodes
+        self._multi_ntype = isinstance(ntype, list)
 
     def __getitem__(self, key):
-        return self._graph._get_n_repr(self._ntid, self._nodes)[key]
+        if self._multi_ntype:
+            ret = {}
+            for (i, ntype) in enumerate(self._ntype):
+                value = self._graph._get_n_repr(self._ntid[i], self._nodes).get(key, None)
+                if value is not None:
+                    ret[ntype] = value
+            return ret
+        else:
+            return self._graph._get_n_repr(self._ntid, self._nodes)[key]
 
     def __setitem__(self, key, val):
-        self._graph._set_n_repr(self._ntid, self._nodes, {key : val})
+        if self._multi_ntype:
+            assert isinstance(val, dict), \
+                'Current HeteroNodeDataView has multiple node types, ' \
+                'please passing the node type and the corresponding data through a dict.'
```

Comment:
Can we allow this by setting the feature to all types.

Reply:
Different types may have different shape. How can we know who is who in this case?